### PR TITLE
test: check for replica attachment creation in dynamically provisioned test

### LINKED ISSUE #221: test: Create a e2e test that checks if the correct number of azvolume replica attachments are created
**Is your feature request related to a problem?/Why is this needed**
Currently, we have no e2e test that asserts that the correct number of replica attachment for a volume is created. As a result, any issues in in the replica creation workflow is discovered during performance testing. Would be good to have a separate test to check this workflow. 

**Describe the solution you'd like in detail**

Create a pod with a volume with maxShares >2 and check that the correct number of replica attachments are created once the pod gets into 'Running" state. 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by kassarl**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Adds a check for the replica attachments being successfully created during e2e test case

**Which issue(s) this PR fixes**:

Fixes #221 

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: test/e2e/dynamic_provisioning_test.go [hunk 1]
```diff
@@ -891,7 +895,7 @@ func (t *dynamicProvisioningTestSuite) defineTests(isMultiZone bool, schedulerNa
 			expectedString = "hello world\r\n"
 		}
 
-		storageClassParameters := map[string]string{"skuName": "StandardSSD_LRS"}
+		storageClassParameters := map[string]string{"skuName": "StandardSSD_LRS", "maxShares": "2"}
```
Comment (edreed) [inline test/e2e/dynamic_provisioning_test.go:898]: I thought we already had another test with replicas (perhaps its with the scheduler extender tests)? We should use that test case instead of changing this one.

File: test/e2e/testsuites/dynamically_provisioned_pod_failover_tester.go [hunk 2]
```diff
@@ -61,6 +68,26 @@ func (t *PodFailover) Run(client clientset.Interface, namespace *v1.Namespace, s
...
+	//changes
+	time.Sleep(1 * time.Minute)
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"disk.csi.azure.com/requested-role": "Replica"}}
+	azVolumeAttachments, err := t.AzDiskClient.DiskV1alpha1().AzVolumeAttachments(consts.AzureDiskCrdNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: labels.Set(labelSelector.MatchLabels).String()})
+	if err != nil {
+		ginkgo.Fail("failed to get replica attachments")
+	}
+	maxShares, err := strconv.ParseInt(t.StorageClassParameters["maxShares"], 10, 0)
+	if len(azVolumeAttachments.Items) != int(maxShares-1) {
+		ginkgo.Fail("incorrect number of replica attachments found")
```
Comment (edreed) [inline test/e2e/testsuites/dynamically_provisioned_pod_failover_tester.go:81]: It might be useful to include the expected and actual number of replicas in the message.

File: test/e2e/testsuites/dynamically_provisioned_pod_failover_tester.go [hunk 3]
```diff
@@ -61,6 +68,26 @@ func (t *PodFailover) Run(client clientset.Interface, namespace *v1.Namespace, s
...
+	}
+	//Check that they were attached
+
+	for _, replica := range azVolumeAttachments.Items {
+		if replica.Status.State != "Attached" {
+			ginkgo.Fail("not all replicas successfully attached")
```
Comment (edreed) [inline test/e2e/testsuites/dynamically_provisioned_pod_failover_tester.go:87]: Include expected and actual replica count in attached state.

File: test/e2e/dynamic_provisioning_test.go (hunk 1 above)
Comment (edreed) [reply test/e2e/dynamic_provisioning_test.go:898]: Ah, so the test case I was thinking of is https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/7e3ae4379cd7a7b8c2bcbca1ed09554098d735e6/test/e2e/azdiskschedulerextender_test.go#L203. However, this one fails over the pod several times during the test. We can either check between each iteration that the replicas have been created, or just do it at the end. This is a good semi-stress test case, but I wonder if we should add a simple test case to verify replica counts.

I believe that we should do the following:
1. Instead of modifying this existing test case, add a new test case that is essentially a duplicate of this one that sets `maxShares` to 2 and check for replicas, e.g. "Should test pod failover with replicas"
2. Update the scheduler extender test case to verify the expected number of replica attachments after the pod failovers have been performed.
